### PR TITLE
DAOS-16153 test: Do not run NLT fi tests for release builds.

### PR DESCRIPTION
**Pull request by ashleypittman**
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
See the ticket for details, I wonder if we're supposed to be using the `release` build type for unit tests even if it is a release build?

File: utils/node_local_test.py [hunk 1]
```diff
@@ -6395,10 +6395,26 @@ def run(wf, args):
...
+
+    if run_fi:
         args.server_debug = 'INFO'
         args.memcheck = 'no'
         args.dfuse_debug = 'WARN'
+
+        fs = subprocess.run([os.path.join(conf['PREFIX'], 'bin', 'fault_status')], check=False)
+        print(fs)
+        if fs.returncode != 0:
+            print("Unable to verify fault injection feature")
```
Comment (brianjmurrell) [inline utils/node_local_test.py:6416]: Duplicated code from lines 6401-6406.  Could this be DRY'd into a small function?

File: utils/node_local_test.py [hunk 2]
```diff
@@ -6395,10 +6395,26 @@ def run(wf, args):
...
     # as a comparison against other builds.
+    run_fi = False
+
     if args.perf_check or fi_test or fi_test_dfuse:
+        fs = subprocess.run([os.path.join(conf['PREFIX'], 'bin', 'fault_status')], check=False)
+        print(fs)
+        if fs.returncode == 0:
+            run_fi = True
+        else:
+            print("Unable to verify fault injection feature")
```
Comment (brianjmurrell) [inline utils/node_local_test.py:6406]:
```suggestion
            print("Unable to verify fault injection feature on release builds")
```

Just so the log reader knows why it was unable and that it's an expected situation.

Comment (ashleypittman) [reply utils/node_local_test.py:6406]: There are other potential reasons for fault injection to be compiled out.

File: utils/node_local_test.py (hunk 1 above)
Comment (ashleypittman) [reply utils/node_local_test.py:6416]: This block should have been removed.  I'll re-push.

**Review by brianjmurrell — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**